Mathlib/Order/Partition/Finpartition.lean

```diff
@@ -511,6 +511,16 @@ theorem part_mem (ha : a ∈ s) : P.part a ∈ P.parts := by simp [part, ha, cho
 
 theorem mem_part (ha : a ∈ s) : a ∈ P.part a := by simp [part, ha, choose_property]
 
+theorem part_surjOn : Set.SurjOn P.part s P.parts := fun p hp ↦ by
+  obtain ⟨x, hx⟩ := P.nonempty_of_mem_parts hp
+  have hx' := mem_of_subset ((le_sup hp).trans P.sup_parts.le) hx
+  use x, hx', (P.existsUnique_mem hx').unique ⟨P.part_mem hx', P.mem_part hx'⟩ ⟨hp, hx⟩
+
+theorem exists_part_bijOn : ∃ r ⊆ s, Set.BijOn P.part r P.parts := by
```

Comment:
I think this is a clearer name, to disambiguate that it's not the part that exists
```suggestion
theorem exists_subset_part_bijOn : ∃ r ⊆ s, Set.BijOn P.part r P.parts := by
```